core/che-core-api-model/src/main/java/org/eclipse/che/api/core/model/workspace/config/MachineConfig.java

```diff
@@ -28,6 +28,13 @@
    */
   String MEMORY_LIMIT_ATTRIBUTE = "memoryLimitBytes";
 
+  /**
+   * Name of the attribute from {@link #getAttributes()} which if present defines memory limit of
+   * the machine in bytes. If memory limit is set in environment specific recipe this attribute used
+   * in {@code MachineConfig} should override value from recipe.
+   */
+  String MEMORY_REQUEST_ATTRIBUTE = "memoryRequestBytes";
```

Comment:
Can you add a comment that infra implementation might not respect this attribute?